src/main/java/cms/gov/madie/measure/resources/AdminController.java

```diff
@@ -198,6 +212,13 @@ public ResponseEntity<Measure> correctMeasureVersion(
       throw new ResourceNotFoundException(error);
     }
 
+    if (!measureToCorrectVersion.getMeasureSet().getOwner().equals(harpId)) {
```

Comment:
The above methods all use the same logic, maybe we could put them together in a small check method?